aws_sagemaker_studio/sagemaker_studio_image_build/xgboost_bring_your_own/Batch_Transform_BYO_XGB.ipynb

```diff
@@ -0,0 +1,1112 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Part 2: Building the Container and Training the model\n",
+    "\n",
+    "\n",
+    "### Step 5: Set up SageMaker Experiments"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this notebook, we first build the Docker image by providing the Dockerfile discussed before and train a model using that Dockerfile\n",
+    "\n",
+    "We use SageMaker Experiments for data scientists to track the lineage of the model from the raw data source to the preprocessing steps and the model training pipeline. With SageMaker Experiments, data scientists can compare, track and manage multiple diferent model training jobs, data processing jobs, hyperparameter tuning jobs and retain a lineage from the source data to the training job artifacts to the model hyperparameters and any custom metrics that they may want to monitor as part of the model training.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a SageMaker Experiment\n",
+    "cc_experiment = Experiment.create(\n",
+    "    experiment_name=f\"CreditCardDefaultBYOI-{int(time.time())}\", \n",
+    "    description=\"Predict credit card default from payments data\", \n",
+    "    sagemaker_boto_client=sm)\n",
+    "print(cc_experiment)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In addition to training, we want to track the lineage of the entire machine learing pipeline also including the processing job above. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Start Tracking parameters used in the Pre-processing pipeline.\n",
+    "with Tracker.create(display_name=\"Preprocessing\", sagemaker_boto_client=sm) as tracker:\n",
+    "    tracker.log_parameters({\n",
+    "        \"train_test_split_ratio\": 0.2,\n",
+    "        \"random_state\":0\n",
+    "    })\n",
+    "    # we can log the s3 uri to the dataset we just uploaded\n",
+    "    tracker.log_input(name=\"ccdefault-raw-dataset\", media_type=\"s3/uri\", value=raw_data_location)\n",
+    "    tracker.log_input(name=\"ccdefault-train-dataset\", media_type=\"s3/uri\", value=train_data_location)\n",
+    "    tracker.log_input(name=\"ccdefault-test-dataset\", media_type=\"s3/uri\", value=test_data_location)\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Step 6: Build XgBoost container for training\n",
+    "\n",
+    "The code for the XGB container is already supplied with this notebook. We simply need to build this container and push it to ECR. The single line of code below will do it."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!sm-docker build ."
```

Comment:
This breaks on the Data Science kernel. Looks like if you switch the to the TensorFlow CPU optimized kernel like used in the other notebook, this will work.

Reply:
Accidentally tried the TF2 CPU opt kernel and it doesn't work there either.

Reply:
@aaronmarkham Thanks for running through the notebooks. The goal of the XGB notebook is to show an end to end flow with the BYO docker capability (which includes steps such as data processing, model training, model tracking, offline inference etc). Perhaps this should be made more clear in the intro. Since the TF notebook is more focused on the "how", this notebook is more focused on the user experience for a typical data scientist. I will run the notebooks in Studio again and make the other fixes. Agree that we probably don't need to upgrade boto3 etc at the beginning.

Reply:
I have fixed this error -- this notebook works with the data science kernel now.